lib/basedriver/helpers.js

```diff
@@ -18,13 +18,14 @@ const CACHED_APPS_MAX_AGE = 1000 * 60 * 60 * 24; // ms
 const APPLICATIONS_CACHE = new LRU({
   maxAge: CACHED_APPS_MAX_AGE, // expire after 24 hours
   updateAgeOnGet: true,
-  dispose: async (app, {fullPath}) => {
-    if (!await fs.exists(fullPath)) {
-      return;
-    }
-
-    logger.info(`The application '${app}' cached at '${fullPath}' has expired`);
-    await fs.rimraf(fullPath);
+  dispose: (app, {fullPath}) => {
+    logger.info(`The application '${app}' cached at '${fullPath}' has ` +
+      `expired after ${CACHED_APPS_MAX_AGE}ms`);
+    setTimeout(0, async () => {
+      if (fullPath && await fs.exists(fullPath)) {
```

Comment:
any exception thrown in here will become an unhandled rejection

Reply:
and why are we using setTimeout anyway?

Reply:
that's ok, both function calls are safe

Reply:
setTimeout is used in order to have async calls

Reply:
why not just

```js
(async() => { /*function body*/ })();
```?

Reply:
dunno, I find setTimeout more obvious

Reply:
hmm. i guess for me there's nothing that prevents you from running an async function just like you would a regular function if you're not going to wait on the result. but it's fine how it is.